client/context/Base.tsx

```diff
@@ -1,83 +1,85 @@
-import React, { createContext, useContext, useEffect, useState } from 'react';
-import { formatClubsData, formatEventsData } from "../utils";
+import React, { createContext, useContext, useEffect, useState } from 'react'
+import { indexData } from '../utils'
+
+export type Id = number
 
 interface AppData {
-    clubs: Map<Id, ClubInfo>;
-    events: Map<Id, EventInfo>;
+  clubs: Map<Id, Club>
+  events: Map<Id, Event>
 }
 
-export interface ClubInfo {
-    name: string
-    description: string
+export type Club = {
+  id: Id
+  name: string
+  description: string
 }
 
-export interface EventInfo {
-    name: string
-    description: string
+export type Event = {
+  id: Id
+  name: string
+  description: string
```

Comment:
omg adding the ID to the Club and Event types simplified so much 🙌, I can't remember why we separated it in the first place..